Qt6 build warns about: QgsGrassDataFile::open hides overloaded virtual function

### What is the bug or the crash?

*as a developer* 
building Qt6 with GRASS8 warns about
```
/root/QGIS/src/providers/grass/qgsgrassdatafile.h:42:10: error: 'QgsGrassDataFile::open' hides overloaded virtual function [-Werror,-Woverloaded-virtual]
    bool open( FILE *fh );
         ^
```
https://github.com/qgis/QGIS/blob/e338d8d0e43b8cef6c999ac1d0181cf43006b3cb/src/providers/grass/qgsgrassdatafile.h#L42
see https://cdash.orfeo-toolbox.org/viewBuildError.php?buildid=94421

### Steps to reproduce the issue

compile `WITH_QT6=ON` and `WITH_GRASS8=ON` same as in #47887 - but set `WERROR=false`

### Versions

2.24.1 and master

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

not asking for support, just an issue tracker for developers

--- a/src/providers/grass/qgsgrassdatafile.h
+++ b/src/providers/grass/qgsgrassdatafile.h
@@ -39,6 +39,7 @@ class GRASS_LIB_EXPORT QgsGrassDataFile : public QFile
   public:
     explicit QgsGrassDataFile( QObject *parent = nullptr );
     // We need FILE* to be able to test feof but QFile::open(FILE *, OpenMode) is not virtual
+    using QFile::open;
     bool open( FILE *fh );
     // Block until all data are read
     qint64 readData( char *data, qint64 len ) override;

